strawberry/lazy_type.py

```diff
@@ -48,12 +48,14 @@ def resolve_type(self) -> Type:
             if main_module.__spec__ and main_module.__spec__.name == self.module:
                 module = main_module
             elif hasattr(main_module, "__file__") and hasattr(module, "__file__"):
-                if (
-                    main_module.__file__
-                    and module.__file__
-                    and Path(main_module.__file__).samefile(module.__file__)
-                ):
-                    module = main_module
+                main_file = main_module.__file__
+                module_file = module.__file__
+                if main_file and module_file:
+                    try:
+                        is_samefile = Path(main_file).samefile(module_file)
+                    except Exception:
```

Comment:
can we be more explicit with the exception? :D 

also I think it might be worth adding a comment on why we do this :)

Reply:
Sure thing! Thanks for already reviewing it